src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2346,11 +2346,19 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 
 	MonoClass *klass = cmethod->klass;
 	MonoType *etype = mono_class_get_context (klass)->class_inst->type_argv [0];
-	gboolean supported = TRUE;
+	gboolean is_primitive_element_type = MONO_TYPE_IS_VECTOR_PRIMITIVE (etype);
 
-	if (!MONO_TYPE_IS_VECTOR_PRIMITIVE (etype))
+	// special case SN_get_IsSupported intrinsic which verifies whether a type parameter T is supported for a generic vector
+	if (id == SN_get_IsSupported) {
+		MonoInst *ins = NULL;
+		EMIT_NEW_ICONST (cfg, ins, is_primitive_element_type ? 1 : 0);
+		return ins;
+	}
+
+	if (!is_primitive_element_type)
 		return NULL;
 
+	gboolean supported = TRUE;
```

Comment:
I would maybe remove the `gboolean supported` variable and replace the `supported = False;` with `return NULL;`  as we now deal with `SN_get_IsSupported` earlier. However, it is just my feeling and feel free to merge.)

Reply:
I agree, thanks!